bindings/postgres/postgres.go

```diff
@@ -0,0 +1,161 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/pkg/errors"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+// List of operations.
+const (
+	execOperation  bindings.OperationKind = "exec"
+	queryOperation bindings.OperationKind = "query"
+	closeOperation bindings.OperationKind = "close"
+
+	connectionURLKey = "url"
+	commandSQLKey    = "sql"
+)
+
+// Postgres represents PostgreSQL output binding
+type Postgres struct {
+	logger logger.Logger
+	db     *pgxpool.Pool
+}
+
+var _ = bindings.OutputBinding(&Postgres{})
+
+// NewPostgres returns a new PostgreSQL output binding
+func NewPostgres(logger logger.Logger) *Postgres {
+	return &Postgres{logger: logger}
+}
+
+// Init initializes the PostgreSql binding
+func (p *Postgres) Init(metadata bindings.Metadata) error {
+	url, ok := metadata.Properties[connectionURLKey]
+	if !ok || url == "" {
+		return errors.Errorf("required metadata not set: %s", connectionURLKey)
+	}
+
+	poolConfig, err := pgxpool.ParseConfig(url)
+	if err != nil {
+		return errors.Wrap(err, "error opening DB connection")
+	}
+
+	p.db, err = pgxpool.ConnectConfig(context.Background(), poolConfig)
+	if err != nil {
+		return errors.Wrap(err, "unable to ping the DB")
+	}
+
+	return nil
+}
+
+// Operations returns list of operations supported by PostgreSql binding
+func (p *Postgres) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{
+		execOperation,
+		queryOperation,
+		closeOperation,
+	}
+}
+
+// Invoke handles all invoke operations
+func (p *Postgres) Invoke(req *bindings.InvokeRequest) (resp *bindings.InvokeResponse, err error) {
+	if req == nil || req.Metadata == nil {
```

Comment:
metadata is not required for close operation.

Reply:
fixed